app/src/main/java/tech/relaycorp/courier/domain/server/ServerConnectionService.kt

```diff
@@ -0,0 +1,64 @@
+package tech.relaycorp.courier.domain.server
+
+import kotlinx.coroutines.flow.asFlow
+import tech.relaycorp.courier.common.PublishChannel
+import tech.relaycorp.courier.data.database.StoredMessageDao
+import tech.relaycorp.courier.data.disk.DiskRepository
+import tech.relaycorp.courier.data.disk.MessageDataNotFoundException
+import tech.relaycorp.courier.data.model.MessageType
+import tech.relaycorp.courier.data.model.StoredMessage
+import tech.relaycorp.courier.data.network.cogrpc.CogRPC
+import tech.relaycorp.courier.data.network.cogrpc.CogRPCServer
+import tech.relaycorp.courier.domain.DeleteMessage
+import tech.relaycorp.courier.domain.StoreMessage
+import tech.relaycorp.courier.domain.client.UniqueMessageId
+import javax.inject.Inject
+
+class ServerConnectionService
+@Inject constructor(
+    private val storeMessage: StoreMessage,
+    private val storedMessageDao: StoredMessageDao,
+    private val diskRepository: DiskRepository,
+    private val deleteMessage: DeleteMessage
+) : CogRPCServer.ConnectionService {
+
+    private val clientConnected = PublishChannel<Unit>()
+    fun clientConnected() = clientConnected.asFlow()
+
+    override suspend fun collectCargo(cca: CogRPC.MessageReceived): Iterable<CogRPC.MessageDelivery> {
+        clientConnected.send(Unit)
```

Comment:
Why do we need to announce when a client connects?

Note that the server can't control the order of the methods used by the client, so even though we certainly expect this to be the first method called by the client, if we needed a reliable way to detect when a new client connects, we'd need to do it in a different way (it'll basically depend on the interface offered by the server).

Reply:
I'll try to gather this information directly from the CogRPC server then.

Reply:
Basically the goal was to know how many clients have synced with the Courier. But, most important is to know if there is someone still connected or not, so I'll focus on that.

Reply:
> Basically the goal was to know how many clients have synced with the Courier

I don't think we can do that without changes to the protocol.

> But, most important is to know if there is someone still connected or not, so I'll focus on that.

That's certainly doable. The exact way to compute that will depend on what the gRPC server gives us:

- They may keep a pool of active connections with each client, in which case it'd be a matter of counting the elements of the pool. I haven't seen this with other gRPC servers though.
- They may emit an event when a new TCP connection is established and another event when a connection ends, so we could keep a counter of those events to calculate the number of open connections. We'd just have to make sure that counter is thread-safe.

Either way, you should be able to call a method that tells you whether there's any open connection and that method could do one of those things behind the scenes.